### PR TITLE
Add gitter sidecar chat to the project project page

### PR DESCRIPTION
**Pull request by dimasciput**
@timlinux 
projecta-gitter

### CODE REVIEW

**Comment by timlinux** [comment]
Hi @dimasciput 

* Any time we change models we should have a migration rule accompanying the change.
* Any time we change models we should have tests added for CRUD so that we can verify the new functionality

Could you take care of the above so we can merge?

Regards

Tim

**Comment by dimasciput** [comment]
@timlinux 

I just updated my code based on your comments, please review it again. 

Thanks.

File: django_project/base/migrations/0002_project_gitter_room.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    dependencies = [
+        ('base', '0001_initial'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='project',
+            name='gitter_room',
+            field=models.CharField(blank=True, max_length=255, null=True, help_text='Gitter room name, e.g. gitterhq/sandbox', validators=[base.models.project.validate_gitter_room_name]),
```
Comment (timlinux) [inline django_project/base/migrations/0002_project_gitter_room.py]: PEP8

According to our coding conventions it should rather be formatted like this:

```
            field=models.CharField(
                blank=True, 
                max_length=255, 
                null=True, 
                help_text='Gitter room name, e.g. gitterhq/sandbox', 
                validators=[base.models.project.validate_gitter_room_name]),
```

Please see https://github.com/inasafe/inasafe/wiki/Coding-Standards for a detailed run through of our expectations with regards to python coding.

File: django_project/base/models/project.py [hunk 2]
```diff
@@ -46,6 +48,22 @@ def get_queryset(self):
                 private=False).filter(approved=True)
 
 
+def validate_gitter_room_name(value):
+    """Ensure user enter proper gitter room name
+
+    :param value: string input
+    :raise ValidationError
```
Comment (timlinux) [inline django_project/base/models/project.py]: :raise: ValidationError

File: django_project/base/migrations/0002_project_gitter_room.py (hunk 1 above)
Comment (timlinux) [inline django_project/base/migrations/0002_project_gitter_room.py]: Oh my apologies - this was generated code by django migrations (shame on them for making on pep8 compliant code :-P) I see lower down you have adopted the correct coding style.

Thanks

Tim

**Comment by timlinux** [comment]
LGTM @dimasciput - just one small formatting error in your doctoring above. Feel free to merge when you are done fixing that.

**Pull request merged**